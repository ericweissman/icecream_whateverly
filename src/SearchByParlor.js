import React, { Component } from 'react';
<<<<<<< HEAD
import App from './App.js';
=======
>>>>>>> 2a92d4c6f21b849ce00ce5bacc1c8b106ee21560
import './css/Main.scss';

const SearchByParlor = (props) => {
    let parlors = props.parlors
    return (
        <div className="SearchByParlor">
            <select className="search-parlor" onChange={props.searchParlor}>
<<<<<<< HEAD
                <option value="" selected disabled hidden>Select Parlor</option>
=======
                <option value="" selected disabled hidden>Select parlor</option>
>>>>>>> 2a92d4c6f21b849ce00ce5bacc1c8b106ee21560
                <option value="show all">Show All Parlors</option>
                {parlors.map((parlor) => {
                    return (
                        <option value={parlor.parlorName}>{parlor.parlorName}</option>
                        )
                    })
                }
            </select>
        </div>
    )
}

export default SearchByParlor;