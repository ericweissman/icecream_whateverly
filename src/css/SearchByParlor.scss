@import '_variables.scss';

.search-parlor {
    border: solid 2px #757575;
    color: #757575;
    height: 47px;
    margin-left: 15px;
    width: 265px;
    padding: 6px;
    text-align: center;
    outline: none;
}