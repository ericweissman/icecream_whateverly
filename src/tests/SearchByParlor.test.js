import React from 'react';
import SearchByParlor from '../SearchByParlor';
import { shallow } from 'enzyme';

const parlors = [
  {
    "parlorName": "Little Man Ice Cream",
    "description": "Quirky shop for exotic ice cream",
    "address": "2620 16th St, Denver, CO 80211",
    "daysOpen": [
      {
        "day": "Sunday",
        "open": "11 AM",
        "close": "10 PM"
      },
      {
        "day": "Monday",
        "open": "11 AM",
        "close": "10 PM"
      },
      {
        "day": "Tuesday",
        "open": "11 AM",
        "close": "10 PM"
      },
      {
        "day": "Wednesday",
        "open": "11 AM",
        "close": "10 PM"
      },
      {
        "day": "Thursday",
        "open": "11 AM",
        "close": "10 PM"
      },
      {
        "day": "Friday",
        "open": "11 AM",
        "close": "11 PM"
      },
      {
        "day": "Saturday",
        "open": "11 AM",
        "close": "11 PM"
      }
    ],
    "phoneNumber": "303-455-3811",
    "website": "http://www.littlemanicecream.com/latest-scoop/menu/",
    "flavors": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27],
    "numberOfFlavors": 27,
    "priceRange": "$",
    "review": 4.7
  }]

const searchParlorMock = jest.fn();

describe('SearchByParlor', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <SearchByParlor parlors={parlors} 
                      searchParlor={searchParlorMock} />
    )
  })

  it('should call searchParlor when the dropdown is manipulated', () => {
    wrapper.find('.search-parlor').simulate('change', { target: { value: '' } });
    expect(searchParlorMock).toBeCalled();
  })
})