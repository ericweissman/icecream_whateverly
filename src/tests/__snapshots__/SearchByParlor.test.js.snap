// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchByParlor should match snapshot when all data is passed correctly 1`] = `
<div
  className="SearchByParlor"
>
  <select
    className="search-parlor"
    onChange={[MockFunction]}
  >
    <option
      disabled={true}
      hidden={true}
      selected={true}
      value=""
    >
      Select Parlor
    </option>
    <option
      value="show all"
    >
      Show All Parlors
    </option>
    <option
      value="Little Man Ice Cream"
    >
      Little Man Ice Cream
    </option>
  </select>
</div>
`;
